.github/workflows/pir_end_to_end_tests.yml

```diff
@@ -0,0 +1,163 @@
+name: PIR E2E Tests
+
+on:
+  workflow_dispatch:
+  schedule:
+     - cron: '0 3 * * 1-5' # 3AM UTC offsetted to legacy to avoid action-junit-report@v4 bug
+  pull_request:
+      
+jobs:
+  pir-e2e-tests:
+    name: PIR e2e tests
+    runs-on: ${{ matrix.runner }}
+    strategy:
+      fail-fast: false
+      matrix:
+        runner: [macos-14-xlarge]
+        include:
+          - xcode-version: "15.4"
```

Comment:
I think we are changing this to 16, so we should check what we have in main when we merge.